src/git/fs.ml

```diff
@@ -635,7 +642,13 @@ module Make (IO: IO) (D: Hash.DIGEST) (I: Inflate.S) = struct
       | Some v -> Some (Cstruct.of_string (Hash.to_hex v))
     in
     let lock = lock_file t r in
-    IO.test_and_set_file file ~temp_dir ~lock ~test:(raw test) ~set:(raw set)
+    let is_read_packed_ref = Reference.Set.mem r t.read_packed_refs in
+    let test = if is_read_packed_ref then None else raw test in
```

Comment:
Does this mean that when updating a packed ref, you can specify *any* value for `test` and it will update? I guess if this code is going away soon it probably doesn't matter, but might be worth a warning comment in the API.

Reply:
This should be fixed now.